app/controllers/scim_v2/groups_controller.rb

```diff
@@ -60,14 +60,26 @@ def create
     end
 
     def replace
+      # HACK to make Keyacloak with https://github.com/mitodl/keycloak-scim work
+      # This SCIM server API does not care about $ref, but having $ref present
+      # forces false positive 400 Bad Request being returned.
+      # Firstly, here https://github.com/pond/scimitar/blob/09b284778340ee067df9e5140d0230386afb9992/app/controllers/scimitar/resources_controller.rb#L176C9-L176C67
+      #   The following error is raised.
+      #   ActiveModel::UnknownAttributeError: unknown attribute '$ref' for Scimitar::ComplexTypes::ReferenceMember.
+      # Secondly, if I add $ref to the schema, then
+      #   scimitar tries to create attr_accessor for $ref, but Ruby does not support methods like that.
+      #   https://github.com/pond/scimitar/blob/09b284778340ee067df9e5140d0230386afb9992/app/models/scimitar/schema/derived_attributes.rb#L14
+      # Therefore, $ref is just removed from payload.
+      params[:members]&.each { |member| member.delete("$ref") }
+
       super do |group_id, scim_resource|
         storage_class.transaction do
           group = storage_scope.find(group_id)
           group.from_scim!(scim_hash: scim_resource.as_json)
           raise_result_errors_for_scim(
             Groups::UpdateService
               .new(user: User.current, model: group)
-              .call(user_ids: scim_resource.members.map(&:value))
+              .call(user_ids: scim_resource&.members&.map(&:value))
```

Comment:
:yellow_circle: How many of these safe navigations are necessary?

Can the `scim_resource` really be `nil`? Or can only the members be nil?